crates/router/src/core/payments/access_token.rs

```diff
@@ -63,6 +64,7 @@ pub fn update_router_data_with_access_token_result<F, Req, Res>(
             call_connector_action
         ),
         (true, payments::CallConnectorAction::Trigger)
+            | (true, payments::CallConnectorAction::HandleResponse(_))
```

Comment:
is this necessary? if so, please ensure that no other flows are broken because of this change